nomos-da/full-replication/src/lib.rs

```diff
@@ -162,15 +167,90 @@ impl certificate::Certificate for Certificate {
     }
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+pub struct VID {
+    id: [u8; 32],
+    metadata: Metadata,
+}
+
+impl certificate::vid::VID for VID {
+    type CertificateId = [u8; 32];
+
+    fn certificate_id(&self) -> Self::CertificateId {
+        self.id
+    }
+}
+
+impl metadata::Metadata for VID {
+    type AppId = [u8; 32];
+    type Index = FRIndex;
+
+    fn metadata(&self) -> (Self::AppId, Self::Index) {
+        (self.metadata.app_id, self.metadata.index)
+    }
+}
+
+impl Hash for VID {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        state.write(<VID as certificate::vid::VID>::certificate_id(self).as_ref());
+    }
+}
+
+impl From<Certificate> for VID {
+    fn from(cert: Certificate) -> Self {
+        // To simulate the propery of aggregate committment + row commitment in Nomos Da Protocol,
+        // when full replication certificate is converted into the VID (which should happen after
+        // the verification in the mempool) the id is set to the blob hash to allow identification
+        // of the distributed data accross nomos nodes.
+        let id = cert.attestations[0].blob_hash();
```

Comment:
Reading this in where we use indexing (`cert.attestations[0]`) because we are supposed to know that this is called when Certificate is already validated. Made me realise that instead of having a `From<Certificate>` maybe we need a `TryFrom<Certificate>` instead.  This is a basic example of our conversation the other day in where it may fail(?).
Another option would be to encode it in the type system something like:
```
pub struct Certificate<Verified>{
   ...
}
``` 
In where we transition to a verified state and we implement the a `vid(&self) -> VID` method just for the verified state.

Im not sure how would that fit with the traits, but I think it should be possible. wdyt @bacv ?

Reply:
`Certificate<Verified>` is a way to go, will create a new issue to track it.